doc/main/reference/assertion_handler.rst

```diff
@@ -0,0 +1,82 @@
+.. _assertion_handler:
+
+Custom Assertion Handler
+========================
+
+.. note::
+    The availability of the extension can be checked with the ``TBB_EXT_CUSTOM_ASSERTION_HANDLER`` macro defined in
+    ``oneapi/tbb/global_control.h`` and ``oneapi/tbb/version.h`` headers.
+
+.. contents::
+    :local:
+    :depth: 2
+
+Description
+***********
+
+OneTBB provides `internal assertion checking
+<https://oneapi-spec.uxlfoundation.org/specifications/oneapi/v1.4-rev-1/elements/onetbb/source/configuration/enabling_debugging_features>`_
+that prints an error message and terminates the application when errors are detected in oneTBB header files or
+the debug version of the library. The custom assertion handler mechanism extends this by allowing developers to
```

Comment:
A few notes:
- the first letter in oneTBB is never capitalized
- there might be assertions in the release version of the library
- I am not sure of the value of embedding the specification link into the text; having it in "See also" seems sufficient to me.